### ISSUE TITLE
[Core] many_tasks regression

### ISSUE THREAD

**Issue by rkooo567** [ray-project/ray#31596]
### What happened + What you expected to happen

https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_jq669a9ga59jtmcp7da8bzs3jw?command-history-section=command_history

many tests recently has failed again. Historically this was always due to worker startup failure. I am going to bisect to find an offending commit. 

### Versions / Dependencies

master

### Reproduction script

n/a

### Issue Severity

High: It blocks me from completing my task.

**Comment by rkooo567** [comment]
Hmm pretty weird... bisect says the first failure commit is the offending commit, but it doesn't make sense at all...

<img width="1325" alt="Screen Shot 2023-01-12 at 6 43 24 AM" src="https://user-images.githubusercontent.com/18510752/212096754-b697162a-1e7b-479c-b414-f478ff4702a3.png">

```
Commit(sha='c4535369d8634b7654e69f4b29b9158dcde267b0', success=False)
Commit(sha='0417cca3d6e948e48295b0dd0e6317e1fda96151', success=True)
Commit(sha='6c0a3c540bd11177d43f08cd3b210eea770e195c', success=None)
Commit(sha='f2ba326d2c54cf7f691d3b2077353e3a17c6203f', success=True)
Commit(sha='6e5bb249c122ec44b1d149c0c033c3297be83da2', success=None)
Commit(sha='3011cd1a5c50d9ce9cb02022575a866247177f43', success=None)
Commit(sha='b8def34be82c902eb0dd01b3bded5617f5ccd7cf', success=True)
Commit(sha='f491e5c4a35390c6eb1a3147b030c90965bc9b57', success=None)
Commit(sha='667b595f700f6bb53f9a669c880fa5fb9a70499c', success=None)
Commit(sha='fe90d7966efcd5f49ad8d63bf8442f83fe69e5d0', success=None)
Commit(sha='0c8b59d2d90df0cfe0f17d9feb7ef9b3e5fe53f2', success=None)
Commit(sha='4e234b7fd741ccb1a24b7e5f0c12cc65be2826c3', success=None)
Commit(sha='f54897accac40a3a688be6567b497caf1b5a2e8a', success=None)
Commit(sha='5c7826f55a0776bcea42b019dd9aec452e729342', success=True)
```
The previous commit 0417cca3d6e948e48295b0dd0e6317e1fda96151 succeeds to run https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_458byqn9sael9ihkb2et7zymhj?command-history-section=command_history

**Comment by rkooo567** [comment]
I will re-run bisect over the night...

**Comment by rkooo567** [comment]
cc @scv119 

Hmm it looks like we have been already pretty close to the failure. I ran bisect

```
Commit(sha='2182af14b147d4e3c973e61ac0238c71ee686667', success=False)
Commit(sha='4a070fc6fb8cece0ce103d6a1e197aec942d78bb', success=False) -> 1723
Commit(sha='296dd272c2867330ef2eef5c82b72abf32dcc8bb', success=True) -> 1755
Commit(sha='15676dda075690f2cb9f83511a18bcde0404f31f', success=True) -> 1761
Commit(sha='b29890d1607378aa144d764635b5dc51bce17150', success=True) -> 1754
```

If the value is < 1750, it is considered as failure. 

I assume we merged a PR that slightly slowed down the worker startup time recently (I couldn't pinpoint it because it is flaky). Given the throughput hasn't been changed 

I'd like to suggest the following changes.

1. Report the CPU utilization as "throughput" (higher is better).
2. Relieve the failure condition. Instead of 1750, we can do something like 1500. We can still detect failures because now we track the performance.
3. Add microbenchmark to test worker startup time. I believe it is planned by @scv119
